### DISCUSSION TITLE
Podman Support?

### DISCUSSION THREAD

**Discussion by Emporea** [nginx-proxy/nginx-proxy#2112] [Q&A] [answered]
Is it possible to use this with podman?
docker gen uses docker.sock to determine container specifications etc.

**Comment by buchdag** [comment]
Unfortunately I have absolutely zero knowledge of the way Podman work so I won't be of much help here.

If Podman doesn't expose an API that is compatible with the Docker Engine API over either a socket, HTTP ou HTTPS it sure won't be possible. There is nothing in docker-gen that would make it specifically compatible with a different API than the Docker Engine API.

**Reply by prashker** [reply]
Yes it is possible - a Docker compatible socket can be exposed by Podman.

**Comment by ian-llewellyn** [comment, marked as answer]
This is old, but still open, so for anyone who's stuck: instead of `-v /var/run/docker.sock:/tmp/docker.sock:ro` use:
```
-v /var/run/podman/podman.sock:/tmp/docker.sock:ro
```

You may have to adjust SELinux to get things working fully.

**Reply by buchdag** [reply]
For rootless podman:
1.
```
systemctl --user enable --now podman.socket
```
2.
```
podman run -d --name nginx-proxy -p 80:80 -v $XDG_RUNTIME_DIR/podman/podman.sock:/tmp/docker.sock:ro docker.io/nginxproxy/nginx-proxy:1.11
```
